app/controllers/incomeSources/add/BusinessAddedObligationsController.scala

```diff
@@ -51,40 +54,70 @@ class BusinessAddedObligationsController @Inject()(authenticate: AuthenticationP
                                                    dateService: DateServiceInterface)
   extends ClientConfirmedController with I18nSupport with FeatureSwitching with IncomeSourcesUtils {
 
-  private def getBusinessNameAndStartDate(incomeSourceId: String)(implicit user: MtdItUser[_]): Option[(String, LocalDate)] = {
-    user.incomeSources.businesses
-      .find(_.incomeSourceId.equals(incomeSourceId))
-      .flatMap { addedBusiness =>
-        for {
-          businessName <- addedBusiness.tradingName
-          startDate <- addedBusiness.tradingStartDate
-        } yield (businessName, startDate)
-      }
+  def getBusinessNameAndStartDate(incomeSource: Either[Throwable, Option[String]])(implicit user: MtdItUser[_]): Either[Throwable, (String, LocalDate)] = {
+    incomeSource match {
+      case Right(Some(incomeSourceId)) =>
+        val addedBusinessDetails = for {
+          addedBusiness <- user.incomeSources.businesses.find(_.incomeSourceId.equals(incomeSourceId))
+          tradeName <- addedBusiness.tradingName
+          tradeStartDate <- addedBusiness.tradingStartDate
+        } yield (tradeName, tradeStartDate)
+
+        val res = addedBusinessDetails match {
+          case Some(businessDetails) => Right((businessDetails._1, businessDetails._2))
```

Comment:
If you can use code like below when dealing with Tuple type, you can alway unapply and extra values you need:
`case Some((businessName, startDate)) => Right((businessName, startDate))`